lux/action/similarity.py

```diff
@@ -39,11 +39,21 @@ def similar_pattern(ldf, intent, topK=-1):
     recommendations : Dict[str,obj]
         object with a collection of visualizations that result from the Similarity action
     """
-    row_specs = list(filter(lambda x: x.value != "", intent))
-    if len(row_specs) == 1:
-        search_space_vc = VisList(ldf.current_vis.collection.copy(), ldf)
 
-        query_vc = VisList(intent, ldf)
+    last = ldf.intent[-1]
+    query = ldf.intent.copy()[0:-1]
+    query.append(lux.Clause(last.attribute, last.attribute, "?"))
+    row_specs = ldf.intent
+    if len(args) == 0:
+        topK = 15
+    else:
+        topK = args[0][0]
```

Comment:
I'm not too sure how the `args` will be formatted, but I'm a little concerned that while `arg[0]` will not be null due to your previous check on length, there might not be a value for `args[0][0]`